dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -509,6 +518,50 @@ private void createBioWorkflowsAndVersionsFromDockstoreYml(List<YamlWorkflow> ya
             LOG.error(message, ex);
             throw new CustomWebApplicationException(message, LAMBDA_FAILURE);
         }
+
+        return updatedWorkflowAndVersions;
+    }
+
+    private void addValidationsToLambdaEvent(LambdaEvent event, Collection<WorkflowAndVersion> workflowAndVersions) {
+
+        String message = workflowAndVersions.stream().map(workflowAndVersion -> {
+            Set<Validation> validations = workflowAndVersion.getVersionValidations();
+            if (isNotEmpty(validations)) {
+                String subMessage = computeMultiValidationMessage(validations);
+                if (StringUtils.isNotEmpty(subMessage)) {
+                    return String.format("In version '%s' of %s '%s': %s", workflowAndVersion.getVersionName(), workflowAndVersion.getWorkflowType().getTerm(), computeWorkflowName(workflowAndVersion), subMessage);
+                }
+            }
+            return null;
+        }).filter(StringUtils::isNotEmpty).collect(Collectors.joining(" "));
+
+        if (StringUtils.isNotEmpty(message)) {
+            event.setMessage(message);
+        }
+    }
+
+    private String computeMultiValidationMessage(Set<Validation> validations) {
+        return validations.stream().filter(v -> !v.isValid()).map(v -> computeValidationMessage(v)).filter(StringUtils::isNotEmpty).collect(Collectors.joining(" "));
+    }
+
+    private String computeValidationMessage(Validation validation) {
+        try {
+            JSONObject json = new JSONObject(validation.getMessage());
+            return json.keySet().stream().map(key -> String.format("file '%s': %s", key, json.get(key))).collect(Collectors.joining(" "));
+        } catch (JSONException ex) {
+            LOG.info("Exception processing validation message JSON", ex);
+            return null;
+        }
+    }
+
+    private String computeWorkflowName(WorkflowAndVersion workflowAndVersion) {
+        String name = workflowAndVersion.getWorkflowName();
+        String repository = workflowAndVersion.getWorkflowRepository();
+        return name != null ? name : repository;
```

Comment:
I would argue that we should still include `repository` when there is a workflow name (like `<repository>/<name>`) so it matches what is displayed in the workflow sidebar. It would also avoid confusion in the (probably rare) case that a workflow name matches a repository name within the same organization